dashboard/app/models/pd/application/teacher1819_application.rb

```diff
@@ -426,6 +427,36 @@ def check_idempotency
       Pd::Application::Teacher1819Application.find_by(user: user)
     end
 
+    def workshop
+      return unless regional_partner
+
+      # If this application is associated with a G3 partner who in turn is
+      # associated with a specific teachercon, return the workshop for that
+      # teachercon
+      if regional_partner.group == 3
+        teachercon = get_matching_teachercon(regional_partner)
+        if teachercon
+          return find_teachercon_workshop(course, teachercon, 2018)
+        end
+      else
+
+      # Default to just assigning whichever of the partner's workshops is
+      # scheduled to start first. We expect to hit this case for G1 and G2
+      # partners, and for any G3 partners without an associated teachercon
+      regional_partner.workshops.order_by_scheduled_start.first
```

Comment:
Don't we need to filter this by course and subject?
And isn't the association [pd_workshops_organized](https://github.com/code-dot-org/code-dot-org/blob/1d61f7a392cbc62d1b1db5d1529f00d5eed71c9b/dashboard/app/models/regional_partner.rb#L36) (rather than `workshops`)?

Reply:
d'oh, yes, that is definitely the association.

The first question is probably one for Tanya; I built it off of [this comment](https://trello.com/c/fYzCpAJb/285-teacher-application-g1-g2-summer-workshop-assignments#comment-5a2744c598f9093aad947cb0) but it does seem like we'll want to do some kind of filtering.

Reply:
I think that comment implies `workshop` to mean of the relevant course (CSD vs. CSP) and subject (5-day summer).